### PR TITLE
[36842] Add pre-commit script to generate ER mermaid diagram from SQL…

### LINKED ISSUE #36842: Generate a mermaid diagram for the database ERD schema
### Description

The docs currently show Airflow's database ERD schema as an image: https://airflow.apache.org/docs/apache-airflow/stable/database-erd-ref.html. To make this searchable, it would be useful to visualize this using mermaid, which also supports ERD schemas: https://mermaid.js.org/syntax/entityRelationshipDiagram.html.

### Use case/motivation

Would be useful to make text in the database diagram searchable.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by dennysreg**
Closes: https://github.com/apache/airflow/issues/36842

Use [paracelsus](https://github.com/tedivm/paracelsus?tab=readme-ov-file#inject-mermaid-diagrams) to generate the `docs/apache-airflow/img/airflow_erd.svg` file based on SQLAlchemy models.

> NOTE: The same tool could be use to generate a mermaid code and inject it to a Markdown file but it seems like rst format is not supported yet.

Before
image

After
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,6 +256,424 @@ following the ASF Policy.
 
   ![Code](https://raw.githubusercontent.com/apache/airflow/main/docs/apache-airflow/img/code.png)
 
+## Database scheme
```
Comment (uranusjr) [inline README.md]: Is it necessary to put this in README? I’m not sure that’s particularly useful TBH…

Comment (dennysreg) [reply README.md]: Not really, we should find a better place to embed it. It was just the first markdown file I found. Any suggestions?

Comment (uranusjr) [reply README.md]: The user-facing documentation already has an image, so maybe somewhere in `contributing-docs`? IIRC there’s a document there on database models.

Comment (dennysreg) [reply README.md]: Unfortunately I can't inject the mermaid in the rst file so I just replace the svg file instead
and restore the original README
https://github.com/dennysreg/airflow/blob/issue-36842-generate-a-mermaid-erd-diagram-from-sqlalchemy-models/docs/apache-airflow/database-erd-ref.rst
image

**Comment by potiuk** [comment]
@dennysreg - do you think you can rebase it and make it ready to be merged?

**Pull request closed without merging**